### ISSUE TITLE
Conventions should only apply their own assembly

### ISSUE THREAD

**Issue by JohnStov** [JohnStov/ReSharperFixieRunner#5]
From @plioi:

I created a sample solution with two test projects: one with no convention classes in it, and one with a custom convention class very similar to the default convention. Each test project had one test class in it, containing a passing test and a failing test.

When the custom convention class exists, I think it was working, but the tests in the other test project, meant to run with the default convention, were no longer recognized as tests (the left-side icons were not present). If I comment out the convention class in full, though, both projects' tests become discoverable again (the left side icons appear).

In other words, the presence of a custom convention class affects more than it should. At the moment, custom convention classes should affect only their containing assembly.

I then modified the custom convention class so that it should have detected zero tests, using a test class naming convention like so: Classes.NameEndsWith("SomethingThatWillNotMatchAnyExistingMethod"). With this convention, methods were still being considered to be tests as if the default convention won. I suspect that the default convention was the convention actually being used, instead of the custom convention.
